tools/rum-models-generator/Sources/CodeGeneration/Print/ObjcInteropPrinter.swift

```diff
@@ -36,15 +36,27 @@ import Foundation
 ///     }
 ///
 public class ObjcInteropPrinter: BasePrinter, CodePrinter {
+    /// List of types which are not exposed to Obj-c.
+    /// Including the nested types of these types.
+    let typesToSkip: Set<String> = [
+        "TelemetryUsageEvent"
+    ]
+
     public init(objcTypeNamesPrefix: String) {
         self.objcTypeNamesPrefix = objcTypeNamesPrefix
     }
 
     // MARK: - CodePrinter
 
     public func print(code: GeneratedCode) throws -> String {
+        let filteredTypes = code.swiftTypes.filter {
+            guard let typeName = $0.typeName else {
+                return true
+            }
+            return !typesToSkip.contains(typeName)
+        }
```

Comment:
suggestion/ Here we're violating OCP by performing domain-specific (usage telemetry) filtering in `ObjcInteropPrinter` that is meant for general purpose (_print objc interop for `GeneratedCode`_).

This can be avoided if this filtering is done at the highest level or `rum-models-generator` CLI:
https://github.com/DataDog/dd-sdk-ios/blob/e5a7b3e0e9dc165a5fd85985dd4e623ab042616e/tools/rum-models-generator/Sources/rum-models-generator/GenerateRUMModels.swift#L70-L73

We could go even further and move this work to `RUMCodeDecorator` and only for objc-interop generation. This is where this filtering should really belong after all.

Reply:
good call, I just made changes to make it configurable via CLI directly